ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -173,22 +178,24 @@ public int execute() {
           return ErrorMsg.REPL_FAILED_WITH_NON_RECOVERABLE_ERROR.getErrorCode();
         }
         Path previousValidHiveDumpPath = getPreviousValidDumpMetadataPath(dumpRoot);
-        boolean isBootstrap = (previousValidHiveDumpPath == null);
-        work.setBootstrap(isBootstrap);
+        work.setBootstrap(previousValidHiveDumpPath == null);
         if (previousValidHiveDumpPath != null) {
           work.setOldReplScope(new DumpMetaData(previousValidHiveDumpPath, conf).getReplScope());
         }
-        //If no previous dump is present or previous dump is already loaded, proceed with the dump operation.
-        if (shouldDump(previousValidHiveDumpPath)) {
-          Path currentDumpPath = getCurrentDumpPath(dumpRoot, isBootstrap);
+        boolean isPrevDumpFailoverReady = checkFailoverStatus(previousValidHiveDumpPath);
```

Comment:
Why does both checkFailoverStatus and getCurrentDumpPath has to handle rollback?

Reply:
checkFailoverStatus is used to examine valid dump dir (containing dump ACK file) whereas in getCurrentDumpPath, we're probing the invalid dump dir (meaning without dump ACK).